vibi-dpu/src/core/review.rs

```diff
@@ -0,0 +1,152 @@
+use std::env;
+
+use serde_json::Value;
+
+use crate::{
+	utils::{hunk::{HunkMap, PrHunkItem}, 
+			review::Review, 
+			gitops::{commit_exists, 
+					git_pull, 
+					get_excluded_files, 
+					generate_diff, 
+					process_diffmap, 
+					generate_blame}}, 
+	db::{hunk::{get_hunk_from_db, store_hunkmap_to_db}, 
+		repo::get_clone_url_clone_dir, 
+		review::{save_review_to_db}},
+	bitbucket::config::get_client};//, core::coverage::process_coverage};
+
+pub async fn process_review(message_data: &Vec<u8>) {
+	let review_opt = parse_review(message_data);
+	if review_opt.is_none() {
+		eprintln!("Unable to deserialize review message");
+		return;
+	}
+	let review = review_opt.expect("review_opt is empty");
+	if hunk_already_exists(&review) {
+		return;
+	}
+	println!("Processing PR : {}", &review.id());
+	commit_check(&review).await;
+	let hunkmap_opt = process_review_changes(&review).await;
+	send_hunkmap(&hunkmap_opt, &review);
+}
+
+fn send_hunkmap(hunkmap_opt: &Option<HunkMap>, review: &Review) {
+	if hunkmap_opt.is_none() {
+		eprintln!("Empty hunkmap in send_hunkmap");
+		return;
+	}
+	let hunkmap = hunkmap_opt.to_owned().expect("empty hunkmap_opt");
+	println!("HunkMap = {:?}", &hunkmap);
+	store_hunkmap_to_db(&hunkmap, review);
+	publish_hunkmap(&hunkmap);
+	// let hunkmap_async = hunkmap.clone();
+	// process_coverage(&hunkmap_async).await; TODO - include in future PR
+}
+
+fn hunk_already_exists(review: &Review) -> bool {
+	let hunk_opt = get_hunk_from_db(&review);
+	if hunk_opt.is_none() {
+		eprintln!("No hunk from get_hunk_from_db");
+		return false;
+	}
+	let hunkmap = hunk_opt.expect("empty hunk from get_hunk_from_db");
+	publish_hunkmap(&hunkmap);
+	println!("Hunk already in db!");
+	return true;
+}
+async fn process_review_changes(review: &Review) -> Option<HunkMap>{
+	let mut prvec = Vec::<PrHunkItem>::new();
+	let fileopt = get_excluded_files(&review);
+	println!("fileopt = {:?}", &fileopt);
+	if fileopt.is_none() {
+		eprintln!("No files to review for PR {}", review.id());
+		return None;
+	}
+	let (_, smallfiles) = fileopt.expect("fileopt is empty");
+	let diffmap = generate_diff(&review, &smallfiles);
+	println!("diffmap = {:?}", &diffmap);
+	let linemap = process_diffmap(&diffmap);
+	let blamevec = generate_blame(&review, &linemap).await;
+	let hmapitem = PrHunkItem::new(
+		review.id().to_string(),
+		review.author().to_string(),
+		blamevec,
+	);
+	prvec.push(hmapitem);
```

Comment:
You only pushing once in this vector. You could have directly defined the variable here instead of creating a mutable.

Reply:
This change has too many cascading effects, will need to change how tmw parses hunks as well